app/src/main/java/org/groundplatform/android/ui/datacollection/tasks/photo/PhotoTaskViewModel.kt

```diff
@@ -58,18 +65,21 @@ constructor(
    * Saves photo data stored on an on-device URI in Ground-associated storage and prepares it for
    * inclusion in a data collection submission.
    */
-  suspend fun savePhotoTaskData(uri: Uri) {
+  fun savePhotoTaskData(uri: Uri) {
     val currentTask = taskWaitingForPhoto
     requireNotNull(currentTask) { "Photo captured but no task waiting for the result" }
+    externalScope.launch(Dispatchers.IO) {
```

Comment:
We spoke, @shobhitagarwal1612 and I agree it would be clearer if the caller switched to IO thread instead of this fun.